app/src/components/change_password.rs

```diff
@@ -53,6 +60,20 @@ fn empty_or_long(value: &str) -> Result<(), validator::ValidationError> {
     }
 }
 
+fn validate_password_complexity(value: &str) -> Result<(), validator::ValidationError> {
+    let has_uppercase = Regex::new(r"[A-Z]").unwrap().is_match(value);
+    let has_lowercase = Regex::new(r"[a-z]").unwrap().is_match(value);
+    let has_symbol = Regex::new(r"[^A-Za-z0-9]").unwrap().is_match(value);
+    let has_number = Regex::new(r"\d").unwrap().is_match(value);
+    let has_length = value.is_empty() || value.len() >= 8;
+
+    if has_uppercase && has_lowercase && has_symbol && has_number && has_length {
+        Ok(())
+    } else {
+        Err(validator::ValidationError::new(""))
```

Comment:
Can we surface which constraint was not fulfilled? Maybe plural.

Reply:
Seems reasonable enough. Stylistically would you rather them each be their own validation function?  E.g. password would get several validators (empty_or_long, has_uppercase, has_lowercase, has_symbol, etc.).  Or do you prefer it to all be in a single function?